datadog_checks_dev/datadog_checks/dev/tooling/manifest_validator/common/validator.py

```diff
@@ -0,0 +1,273 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import abc
+import os
+from typing import Dict
+
+import six
+
+from datadog_checks.dev.tooling.git import content_changed
+from datadog_checks.dev.tooling.utils import get_metadata_file, has_logs, is_metric_in_metadata_file, read_metadata_rows
+
+
+class ValidationResult(object):
+    def __init__(self):
+        self.failed = False
+        self.fixed = False
+        self.messages = {'success': [], 'warning': [], 'failure': [], 'info': []}
+
+    def __str__(self):
+        return '\n'.join(['\n'.join(messages) for messages in self.messages.values()])
+
+    def __repr__(self):
+        return str(self)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseManifestValidator(object):
+    def __init__(
+        self,
+        is_extras=False,
+        is_marketplace=False,
+        check_in_extras=True,
+        check_in_marketplace=True,
+        ctx=None,
+        version="v1",
+        skip_if_errors=False,
+    ):
+        self.result = ValidationResult()
+        self.is_extras = is_extras
+        self.is_marketplace = is_marketplace
+        self.check_in_extras = check_in_extras
+        self.check_in_markeplace = check_in_marketplace
+        self.ctx = ctx
+        self.version = version
+        self.skip_if_errors = skip_if_errors
+
+    def should_validate(self):
+        if not self.is_extras and not self.is_marketplace:
```

Comment:
A docstring may help here since there's a bit of a split in the logic? It looks like we always validate core but there's a flag to enable/disable marketplace/extras?